src/reactive-membrane.ts

```diff
@@ -110,36 +107,27 @@ export class ReactiveMembrane {
         return unwrap(p);
     }
 
-    private getReactiveState(value: any, distortedValue: any): ReactiveState {
-        const {
-            objectGraph,
-        } = this;
-        let reactiveState = objectGraph.get(distortedValue);
-        if (reactiveState) {
-            return reactiveState;
+    private getReactiveHandler(value: any, distortedValue: any): any {
+        let proxy = this.reactiveObjectGraph.get(distortedValue);
+        if (isUndefined(proxy)) {
+            // caching the proxy after the first time it is accessed
+            const handler = new ReactiveProxyHandler(this, distortedValue);
+            proxy = new Proxy(createShadowTarget(distortedValue), handler);
+            registerProxy(proxy, value);
+            this.reactiveObjectGraph.set(distortedValue, proxy);
         }
-        const membrane = this;
-        reactiveState = {
-            get reactive() {
-                const reactiveHandler = new ReactiveProxyHandler(membrane, distortedValue);
-                // caching the reactive proxy after the first time it is accessed
-                const proxy = new Proxy(createShadowTarget(distortedValue), reactiveHandler);
-                registerProxy(proxy, value);
-                ObjectDefineProperty(this, 'reactive', { value: proxy });
-                return proxy;
-            },
-            get readOnly() {
-                const readOnlyHandler = new ReadOnlyHandler(membrane, distortedValue);
-                // caching the readOnly proxy after the first time it is accessed
-                const proxy = new Proxy(createShadowTarget(distortedValue), readOnlyHandler);
-                registerProxy(proxy, value);
-                ObjectDefineProperty(this, 'readOnly', { value: proxy });
-                return proxy;
-            }
-        } as ReactiveState;
-
-        objectGraph.set(distortedValue, reactiveState);
-        return reactiveState;
+        return proxy;
     }
 
+    private getReadOnlyHandler(value: any, distortedValue: any): any {
+        let proxy = this.readOnlyObjectGraph.get(distortedValue);
+        if (isUndefined(proxy)) {
+            // caching the proxy after the first time it is accessed
+            const handler = new ReadOnlyHandler(this, distortedValue);
+            proxy = new Proxy(createShadowTarget(distortedValue), handler);
+            registerProxy(proxy, value);
+            this.readOnlyObjectGraph.set(distortedValue, proxy);
```

Comment:
Any specific reason to use the `distortedValue` as the key instead of the `value`?

IIRC, the original implementation does not have an invariant that `two values cannot produce the same distorted value`.

Reply:
@ravijayaramappa Correct, yeah, it's the same as before.

I'm not really sure how the distortion works, though. In LWC, we [don't do any distortion](https://github.com/salesforce/lwc/blob/d93e2963d8d32b138ce9ed130d98e7bef84b3d74/packages/%40lwc/engine-core/src/framework/membrane.ts#L12-L14), and the tests in this repo don't do a lot of distortion either. Arguably we could release a breaking change for this library to just get rid of the distortion entirely and use the raw `value` everywhere (assuming there aren't any major consumers of this library that are relying on the distortion).

Reply:
I'm ok with removing distortions.